src/Tasks/CodeTaskFactory.cs

```diff
@@ -851,20 +847,44 @@ private Assembly CompileInMemoryAssembly()
         /// <summary>
         /// Combine our default referenced assemblies with those explicitly specified
         /// </summary>
-        private void CombineReferencedAssemblies(List<string> finalReferenceList)
+        private List<string> CombineReferencedAssemblies()
         {
-            foreach (string defaultReference in DefaultReferencedAssemblies)
+            List<string> finalReferenceList = new List<string>();
```

Comment:
super-nit: Since you refactored this and are creating the list here, it may be worth starting with the expected capacity: `s_defaultReferencedFrameworkAssemblyNames.Length + 2 + _referencedAssemblies.Count`.